src/test/groovy/graphql/validation/rules/NoUnusedVariablesTest.groovy

```diff
@@ -89,8 +85,33 @@ class NoUnusedVariablesTest extends Specification {
 
         then:
         errorCollector.containsValidationError(ValidationErrorType.UnusedVariable)
-
     }
 
+    def "variables not used in fragments"() {
+        given:
+        def query = """
+        fragment FragA on Type {
+            field(a: \$a) {
+                ...FragB
+            }
+        }
+        fragment FragB on Type {
+            field(b: \$b) {
+                ...FragC
+            }
+        }
+        fragment FragC on Type {
+            __typename
+        }
+        query Foo(\$a: String, \$b: String, \$c: String) {
+            ...FragA
+        }
+        """
+
+        when:
+        traverse(query)
 
+        then:
+        errorCollector.containsValidationError(ValidationErrorType.UnusedVariable)
+    }
```

Comment:
were we missing a test for this case?

Reply:
I didn't see one; the `variables not used` test in this file is close but doesn't exercise fragments, which seems like a potential blindspot.